test/functional/p2p-fingerprint.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test various fingerprinting protections.
+
+If an orphan block more than a month old or its header are requested by a peer,
+the node should pretend that it does not have it to avoid fingerprinting.
+"""
+
+import time
+
+from test_framework.blocktools import (create_block, create_coinbase)
+from test_framework.mininode import (
+    CInv,
+    NetworkThread,
+    NodeConn,
+    NodeConnCB,
+    msg_headers,
+    msg_block,
+    msg_getdata,
+    msg_getheaders,
+    wait_until,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    p2p_port,
+)
+
+class P2PFingerprintTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    # Build a chain of blocks on top of given one
+    def build_chain(self, nblocks, prev_hash, prev_height, prev_median_time):
```

Comment:
Determine `prev_height` and `prev_median_time` here and simplify code below.

Reply:
I don't understand. Example code?

Reply:
I mean something like:
```python
def build_chain(self, nblocks, prev_hash):
    header = self.nodes[0].getblockheader(prev_hash)
    height = header["height"]
    cur_time = header["mediantime"]

    blocks = []
    for _ in range(nblocks):
        height += 1
        cur_time += 1
        
        coinbase = create_coinbase(height)
        block = create_block(int(prev_hash, 16), coinbase, cur_time)
        block.solve()

        blocks.append(block)
        prev_hash = block.hash
    return blocks

...

    new_blocks = self.build_chain(5, block_hashes[height - 3])
```
This way height below is not needed.